supernode/daemon/mgr/SchedulerMgr.go

```diff
@@ -0,0 +1,33 @@
+package mgr
+
+import (
+	"context"
+
+	"github.com/dragonflyoss/Dragonfly/apis/types"
+)
+
+// SchedulerMgr has two main responsibilities:
+// 1. Responsible for maintaining the correspondence between peer and pieces.
+// 2. Responsible for calculating scheduling results according to certain rules.
+type SchedulerMgr interface {
+	// InitProgress init the correlation information between peers and pieces, etc.
+	InitProgress(ctx context.Context, taskID, cid string) error
+
+	// UpdateProgress update the correlation information between peers and pieces.
+	// 1. update the info about srcCid to tell the scheduler that corresponding peer has the piece now.
+	// 2. update the info about dstCid to tell the scheduler that someone has downloaded the piece form here.
+	// Scheduler will calculate the load and times of error/success for every peer to make better decisions.
+	UpdateSrcProgress(ctx context.Context, taskID, srcCid, dstCid string, pieceRange string, pieceStatus string) error
```

Comment:
I am not sure that pieceRange and pieceStatus can be just strings..

I also doubt whether `InitProgress` function is necessary. Can we do the check here(`UpdateSrcProgress`): if the progress is not tracked previously then initialize it.

Reply:
1. For pieceRange the format `pieceRange` should be like : `start-end`.
For `PieceStatus`, using strings is so flexible, but if we define a corresponding structure for every status such as `CDNStatus, PieceStatus, etc` is redundant. Any good ideas?
2. I think `InitProgress` and `UpdateProgress` is all needed,  they do the different things separately. Otherwise, `UpdateProgress` have to do so much things.